### DISCUSSION TITLE
Verifying each torrent each time it is relaunched

### DISCUSSION THREAD

**Discussion by Qhilm** [haugene/docker-transmission-openvpn#1990] [Q&A] [answered]
Hello,

somehow the container insists on reverifying every torrent each time it's relaunched. I'm on dev branch due to issue fixed in commit 8f0a92a2ff365521dfc1f4c529777017145fb0cf.

Is this expected?

**Comment by pkishino** [comment, marked as answer]
That fix is already in master so dev branch not needed.. are these torrents you added previously on transmission 3.0? Or also ones that you added since upgrading to 4.0? 
Please see the other threads regarding some expected issues between the two versions. It also details how to get back to 3.0 if necessary.
Also, any reason you relaunch it frequently?

**Comment by Qhilm** [comment]
So, if I use docker-transmission-openvpn:latest in Docker Compose, I hit the issue fixed in commit 8f0a92a2ff365521dfc1f4c529777017145fb0cf. Is that not the master branch?

I will check again is the most recent torrent also have the same issue of being reverified next time it relaunches, so I can see if 4.0 torrent are also affected or only 3.0 ones. Currently I'm waiting for verification to complete.

I do not relaunch the container manually, it just crashes on a regular basis. After too many crashes, it is not relaunched again. And next time I relaunch it manually, it reverifies everything. But I always had the problem that this container crashed regularly, on version 3 as well, it's not specific to version 4, and it has never caused all torrent to be reverified each time.

**Comment by pkishino** [comment]
You might want to look into why it is crashing, that is not normal at all.
Exactly why it is verifying I have no idea, that would be better suited to
the transmission forum..

**Comment by Qhilm** [comment]
So I checked, only older torrents seem affected. Recent ones are not. While not 100% sure it does look like the ones created under 4.0 do not go in "verification" mode while the older ones do.

It also seem I cannot remove the older ones. They will seem to remove just fine, but at next relaunch of the container, they will start redownloading from scratch...

I probably should restart from scratch, something seems corrupted.

**Reply by pkishino** [reply]
It is not enough to remove in the UI, you need purge them from the actual file system under transmission-home/resume

**Reply by pkishino** [reply]
This has to do with the change in hashing used between the two versions, this is mentioned in the main thread about this

**Reply by Qhilm** [reply]
> It is not enough to remove in the UI, you need purge them from the actual file system under transmission-home/resume

I'm using the option "trash torrent and files" and if the torrent file was deleted transmission couldn't start downloading it again, right?

I think something is seriously corrupted because I can't even delete torrents anymore from the GUI.

**Reply by pkishino** [reply]
Nope, that’s not enough.. you need to go to the transmission-home folder and clear out the resumes folder.. you can’t do this from the gui

**Comment by Qhilm** [comment]
In the end I deleted everything and now it's stable, using the dev branch.

Does the :latest branch now include 8f0a92a2ff365521dfc1f4c529777017145fb0cf? It was not 20 days ago, I had exactly the corresponding error message in the logs.

**Reply by pkishino** [reply]
If you are using latest it is latest, that is if you always pull before creating the container..

**Reply by Qhilm** [reply]
ok, I found the issue. I have ourobouros updating images, but it only updates images which are in use. Hence, every time I switched back to transmission-openvpn:latest, it would start using the one from sept 4th, which had the issue. I should have manually pulled before switching back, as you said. Now it works fine, I switched to :latest and no torrents are being reverified, all good.

**Reply by pkishino** [reply]
Great

**Comment by balupton** [comment]
So I have a large amount of torrents, however the container restarts before alll verifications are complete - which, upon restart, rather than reverifying only those it had yet to check, it for some reason, reverifies every torrent yet again. Is it possible to have it so it only reverifies that which it has not already checked.

**Reply by pkishino** [reply]
That’s a question for transmission devs…
https://github.com/transmission/transmission
If you do get an answer I’d be happy to hear it and update the documentation if necessary with :)
